docs/dev/Pagination-v2.md

```diff
@@ -98,59 +95,54 @@ The response object has the following format:
 
 ## OpenSearch Data Retrieval Strategy
 
-OpenSearch provides several data retrival APIs that are optimized for different use cases.
+OpenSearch provides several data retrieval APIs that are optimized for different use cases.
 
 At this time, SQL plugin uses simple search API and scroll API.
 
 Simple retrieval API returns at most `max_result_window` number of documents.  `max_result_window` is an index setting.
 
 Scroll API requests returns all documents but can incur high memory costs on OpenSearch coordination node.
 
-Efficient implementation of pagination needs to be aware of retrival API used. Each retrieval strategy will be considered separately.
+Efficient implementation of pagination needs to be aware of retrieval API used. Each retrieval strategy will be considered separately.
 
 The discussion below uses *under max_result_window* to refer to scenarios that can be implemented with simple retrieval API and *over max_result_window* for scenarios that require scroll API to implement.
 
 ## SQL Node Load Balancing
 
-V2 SQL engine supports *sql node load balancing* -- a cursor request can be routed to any SQL node in a cluster. This is achieved by encoding all data necessary to retrieve the next page in the `cursor_id`.
+V2 SQL engine supports *sql node load balancing* &mdash; a cursor request can be routed to any SQL node in a cluster. This is achieved by encoding all data necessary to retrieve the next page in the `cursor_id` property in the response.
 
 ## Feature Design
+To support pagination, v2 SQL engine needs to:
+1. in REST front-end:
+    1. Route supported paginated query to v2 engine for
+        1. Initial requests,
+        2. Next page requests.
+    2. Fallback to v1 engine for queries not supported by v2 engine.
+    3. Create correct JSON response from execution of paginated physical plan by v2 engine.
+2. during query planning:
+    1. Differentiate between paginated and normal query plans.
+    2. Push down pagination to table scan.
+    3. Create paginated physical plan from cursor id.
+3. during query execution:
+    1. Execute paginated physical plan.
+4. in OpenSearch data source: 
+    1. Support pagination push down.
+    2. Support other push down optimizations with pagination.
+
+### Query Plan Changes
+
+All three kinds of requests &mdash; non-paged, initial page, or subsequent page  &mdash; are processed in the same way. Simplified workflow of query plan processing is shown below for reference.
```

Comment:
```suggestion
All three kinds of requests &mdash; non-paged, initial page, or subsequent page  &mdash; are processed at a high-level in the same way. Simplified workflow of query plan processing is shown below for reference.
```

Reply:
We want to say that all the requests go through the same high-level steps, although the details/execution of each step may differ.